### DISCUSSION TITLE
b'{"status":{"error":"Wrong input: Conversion between sparse and regular vectors failed"},"time":0.010832087}'

### DISCUSSION THREAD

**Discussion by HarshSonaiya** [qdrant/qdrant#5062] [General]
def create_sparse_vector(docs: Document, sparse_text_embedding_model) -> SparseVector:

    embeddings = list(sparse_text_embedding_model.embed([docs.page_content]))[0]
    logging.debug(f"Sparse embeddings: {embeddings}")

    if hasattr(embeddings, 'indices') and hasattr(embeddings, 'values'):
        return SparseVector(
            indices=embeddings.indices.tolist(),
            values=embeddings.values.tolist()
        )
    else:
        raise ValueError("The embeddings object does not have 'indices' and 'values' attributes.")
       
def create_sparse_collection(client: QdrantClient):
    """
    Create a collection for sparse vectors in Qdrant if it does not exist.

    Args:
        client (QdrantClient): An instance of QdrantClient.
    """
    if not client.collection_exists(collection_name=sparse_collection):
        client.create_collection(
            collection_name=sparse_collection,
            vectors_config= {},
            sparse_vectors_config={
                "sparse": models.SparseVectorParams(),
            }
        )

    logging.info(f"Created sparse vector collection '{sparse_collection}' in Qdrant.")

 def index_documents(file: str):
 
    # Initialize client and embedding models
    client = create_qdrant_client()

    # Extract and split PDF content
    documents = extract_content_from_pdf(file)

    # Create Qdrant collection
    create_sparse_collection(client)

    # Index sparse embeddings into the sparse collection
    sparse_points = []
    for i, doc in enumerate(tqdm(documents, total=len(documents))):
        sparse_vector = create_sparse_vector(doc, sparse_embedding_model)
        sparse_point = models.PointStruct(
            id=i,
            vector={"sparse":sparse_vector}
        )
        # logging.info(f"Sparse points:{sparse_vector}")
        sparse_points.append(sparse_point)
    logging.info(type(sparse_points[0]),len(sparse_points))
    client.upsert(
        collection_name=sparse_collection,
        points=sparse_points
    )
    logging.info(f"Upserted {len(sparse_points)} points with sparse vectors into Qdrant.")

**Comment by Anush008** [comment]
From the error, it looks like the collection was created with dense vectors earlier.
You can try deleting the collection and re-running
> def create_sparse_collection(client: QdrantClient):

so it's created with the appropriate config.
